frame/airdrop/src/benchmarking.rs

```diff
@@ -0,0 +1,249 @@
+}
+
+pub fn ethereum_proof<T>(
+	ethereum_account: &EthereumKey,
+	reward_account: AccountIdOf<T>,
+) -> ProofOf<T>
+where
+	T: Config<RelayChainAccountId = [u8; 32]>,
+{
+	let msg = hash::<Keccak256>(
+		&signature_verification::ethereum_signable_message(
+			PROOF_PREFIX,
+			&reward_account.using_encoded(|x| hex::encode(x).as_bytes().to_vec()),
+		)[..],
+	);
+	let (sig, recovery_id) =
+		libsecp256k1::sign(&libsecp256k1::Message::parse(&msg), ethereum_account);
+	let mut recovered_signature = [0_u8; 65];
+
+	recovered_signature[0..64].copy_from_slice(&sig.serialize()[..]);
+	recovered_signature[64] = recovery_id.serialize();
+	Proof::Ethereum(EcdsaSignature(recovered_signature))
+}
+
+pub fn ethereum_public(secret: &EthereumKey) -> libsecp256k1::PublicKey {
+	libsecp256k1::PublicKey::from_secret_key(secret)
+}
+
+pub fn cosmos_address(pub_key_slice: &[u8]) -> CosmosPublicKey {
+	let mut pub_key: [u8; 33] = [0; 33];
+	pub_key.copy_from_slice(pub_key_slice);
+	CosmosPublicKey::Secp256r1(pub_key)
+}
+
+pub fn ethereum_address(secret: &EthereumKey) -> EthereumAddress {
+	let mut res = EthereumAddress::default();
+	res.0
+		.copy_from_slice(&hash::<Keccak256>(&ethereum_public(secret).serialize()[1..65])[12..]);
+	res
+}
+
+pub fn cosmos_generate<T>(count: u64) -> Vec<(AccountIdOf<T>, Identity)>
+where
+	T: Config<RelayChainAccountId = [u8; 32]>,
+{
+	let seed: u128 = 12345678901234567890123456789012;
+	(0..count)
+		.map(|i| {
+			let account_id = account("recipient", i as u32, 0xCAFEBABE);
+			(
+				account_id,
+				Identity::Cosmos(
+					SigningKey::from_bytes(
+						&[(&(seed + i as u128)).to_le_bytes(), (&(seed + i as u128)).to_le_bytes()]
+							.concat(),
+					)
+					.unwrap(),
+				),
+			)
+		})
+		.collect()
+}
+
+#[allow(clippy::disallowed_methods)] // Allow unwrap
+pub fn ethereum_generate<T>(count: u64) -> Vec<(AccountIdOf<T>, Identity)>
+where
+	T: Config<RelayChainAccountId = [u8; 32]>,
+{
+	(0..count)
+		.map(|i| {
+			let account_id = account("recipient", i as u32, 0xCAFEBABE);
+			(
+				account_id,
+				Identity::Ethereum(
+					EthereumKey::parse(&hash::<Keccak256>(&i.to_le_bytes())).unwrap(),
+				),
+			)
+		})
+		.collect()
+}
+
+/// `count % 2 == 0` should hold for all x
+pub fn generate_accounts<T>(count: u64) -> Vec<(AccountIdOf<T>, Identity)>
+where
+	T: Config<RelayChainAccountId = [u8; 32]>,
+{
+	assert!(count % 2 == 0, "`x % 2 == 0` should hold for all x");
+	let mut x = cosmos_generate::<T>(count / 2);
+	let mut y = ethereum_generate::<T>(count / 2);
+	x.append(&mut y);
+	x
+}
+
+pub fn hash<T>(input: &[u8]) -> [u8; 32]
+where
+	T: Hasher + Default,
+{
+	let mut hasher: T = Default::default();
+	let mut hash: [u8; 32] = [0; 32];
+
+	hasher.update(input);
+	hash.copy_from_slice(hasher.finalize());
+	hasher.reset();
+
+	hash
+}
+
+benchmarks! {
+	where_clause {
+		where
+			T: Config<RelayChainAccountId = [u8; 32]>,
+			BalanceOf<T>: From<u128>,
+	}
+
+	create_airdrop_benchmark {
+		let x in 100..1000;
```

Comment:
looks like, `x` doesn't used here, and other benchmarks
can we remove it?

Reply:
Shouldn't the linter catch that?

Reply:
https://github.com/ComposableFi/composable/blob/main/.github/workflows/check.yml#L207
We don't check with all features, perhaps why it's not picking it up

Reply:
here uses `benchmarks!` macro
it looks so after the macro expand
```
  const _: () = {
        #[allow(non_camel_case_types)]
        impl<T: Config> ::codec::Encode for Call<T> {
            fn encode_to<__CodecOutputEdqy: ::codec::Output + ?::core::marker::Sized>(
                &self,
                __codec_dest_edqy: &mut __CodecOutputEdqy,
            ) {
                match *self {
                    Call::create_airdrop {
                        ref start_at,
                        ref vesting_schedule,
                    } => {
                        __codec_dest_edqy.push_byte(0usize as ::core::primitive::u8);
                        ::codec::Encode::encode_to(start_at, __codec_dest_edqy);
                        ::codec::Encode::encode_to(vesting_schedule, __codec_dest_edqy);
                    }
                    Call::add_recipient {
                        ref airdrop_id,
                        ref recipients,
                    } => {
                        __codec_dest_edqy.push_byte(1usize as ::core::primitive::u8);
                        ::codec::Encode::encode_to(airdrop_id, __codec_dest_edqy);
                        ::codec::Encode::encode_to(recipients, __codec_dest_edqy);
                    }
                    Call::remove_recipient {
                        ref airdrop_id,
                        ref recipient,
                    } => {
                        __codec_dest_edqy.push_byte(2usize as ::core::primitive::u8);
                        ::codec::Encode::encode_to(airdrop_id, __codec_dest_edqy);
                        ::codec::Encode::encode_to(recipient, __codec_dest_edqy);
                    }
                    Call::enable_airdrop { ref airdrop_id } => {
                        __codec_dest_edqy.push_byte(3usize as ::core::primitive::u8);
                        ::codec::Encode::encode_to(airdrop_id, __codec_dest_edqy);
                    }
                    Call::disable_airdrop { ref airdrop_id } => {
                        __codec_dest_edqy.push_byte(4usize as ::core::primitive::u8);
                        ::codec::Encode::encode_to(airdrop_id, __codec_dest_edqy);
                    }
                    Call::claim {
                        ref airdrop_id,
                        ref reward_account,
                        ref proof,
                    } => {
                        __codec_dest_edqy.push_byte(5usize as ::core::primitive::u8);
                        ::codec::Encode::encode_to(airdrop_id, __codec_dest_edqy);
                        ::codec::Encode::encode_to(reward_account, __codec_dest_edqy);
                        ::codec::Encode::encode_to(proof, __codec_dest_edqy);
                    }
                    _ => (),
                }
            }
        }
        impl<T: Config> ::codec::EncodeLike for Call<T> {}
    };
```
i.e. the macro removes `x`

Reply:
78c5f8d54b7c8754cfbd17b7667172cab465c75b